packages/browser/src/experimentClient.ts

```diff
@@ -147,8 +147,12 @@ export class ExperimentClient implements Client {
     if (!this.user) {
       return this.user;
     }
-    const userPropertiesCopy = { ...this.user.user_properties };
-    return { ...this.user, user_properties: userPropertiesCopy };
+    if (this.user?.user_properties) {
+      const userPropertiesCopy = { ...this.user.user_properties };
+      return { ...this.user, user_properties: userPropertiesCopy };
+    } else {
+      return { ...this.user };
```

Comment:
Does `user_properties` change anything?

It's valid to spread a null or undefined variable, you just won't get any properties. So the before and after here are mostly functionally the same.

Reply:
This was added to make testing easier. Comparing the original user (null/undefined user_properties) and the one as a result of `get/setUser()`would not be equal.